crates/sui-sdk/examples/sign_tx_guide.rs

```diff
@@ -0,0 +1,149 @@
+        )
+        .unwrap(),
+    );
+    let _skp_import_no_flag_2 = SuiKeyPair::Ed25519(
+        Ed25519KeyPair::from_bytes(
+            &Base64::decode("1GPhHHkVlF6GrCty2IuBkM+tj/e0jn64ksJ1pc8KPoI=").unwrap(),
+        )
+        .unwrap(),
+    );
+
+    // import a keypair from a base64 encoded 33-byte `flag || private key`. The signature scheme is determined by the flag.
+    let _skp_import_with_flag_0 =
+        SuiKeyPair::decode_base64("ANRj4Rx5FZRehqwrctiLgZDPrY/3tI5+uJLCdaXPCj6C").unwrap();
+    let _skp_import_with_flag_1 =
+        SuiKeyPair::decode_base64("AdRj4Rx5FZRehqwrctiLgZDPrY/3tI5+uJLCdaXPCj6C").unwrap();
+    let _skp_import_with_flag_2 =
+        SuiKeyPair::decode_base64("AtRj4Rx5FZRehqwrctiLgZDPrY/3tI5+uJLCdaXPCj6C").unwrap();
+
+    // replace `skp_determ_0` with the variable names above
+    let pk = skp_determ_0.public();
+    let sender = SuiAddress::from(&pk);
+    println!("Sender: {:?}", sender);
+
+    // make sure the sender has a gas coin as an example.
+    let coin = sui_client
+        .coin_read_api()
+        .get_coins(sender, None, None, None)
+        .await?
+        .data
+        .into_iter()
+        .next();
+
+    if coin.is_none() {
+        println!("gas coin not found, refer to this re getting gas coin https://docs.sui.io/guides/developer/getting-started/get-coins");
+        return Ok(());
+    }
+    let gas_coin = coin.unwrap();
+
+    // construct an example programmable transaction.
+    let pt = {
+        let mut builder = ProgrammableTransactionBuilder::new();
+        builder.pay_sui(vec![sender], vec![1]).unwrap();
+        builder.finish()
+    };
+
+    let gas_budget = 5_000_000;
+    let gas_price = sui_client.read_api().get_reference_gas_price().await?;
+
+    // create the transaction data that will be sent to the network.
+    let tx_data = TransactionData::new_programmable(
+        sender,
+        vec![gas_coin.object_ref()],
+        pt,
+        gas_budget,
+        gas_price,
+    );
+
+    // derive the digest that the keypair should sign on.
+    let intent_msg = IntentMessage::new(Intent::sui_transaction(), tx_data);
+    let raw_tx = bcs::to_bytes(&intent_msg).expect("bcs should not fail");
+    let mut hasher = sui_types::crypto::DefaultHash::default();
+    hasher.update(raw_tx.clone());
+    let digest = hasher.finalize().digest;
+
+    // use SuiKeyPair to sign the digest.
+    let sui_sig = skp_determ_0.sign(&digest);
+
+    // execute the transaction.
+    let transaction_response = sui_client
+        .quorum_driver_api()
+        .execute_transaction_block(
+            sui_types::transaction::Transaction::from_generic_sig_data(
+                intent_msg.value,
+                Intent::sui_transaction(),
+                vec![GenericSignature::Signature(sui_sig)],
+            ),
+            SuiTransactionBlockResponseOptions::default(),
+            None,
+        )
+        .await?;
+
+    println!(
```

Comment:
Now that we have improved the display of transactions, I think we could do
```suggestion
    println!("{transaction_response}");
```